File: LibraryManagementSystem/Frontend/library-management-system-web/src/app/entities/dtos/library/book-dto.ts
import { PublisherDto } from "./publisher-dto";
import { CategoryDto } from "./category-dto";
import { AuthorDto } from "./author-dto";
import { SubCategoryDto } from "./sub-category-dto";

export interface BookDto {
    bookId: number;
    iSBN10: string | null;
    iSBN13: string | null;
    classification: string | null;
    title: string | null;
    description: string | null;
    publicationYear: number;
    image: string | null;
    isActive: boolean;
    numberOfCopies: number;
    borrowedCopies: number;
    isAvailable: boolean;
    publisher: PublisherDto | null;
    category: CategoryDto | null;
    authors: AuthorDto[] | null;
    subCategories: SubCategoryDto[] | null;
}

File: LibraryManagementSystem/Backend/LibraryManagementSystem/LibraryManagementSystem.Entities/Dtos/Library/BookDto.cs
namespace LibraryManagementSystem.Entities.Dtos.Library
{
    public class BookDto
    {
        public int BookId { get; set; }
        public string? ISBN10 { get; set; }
        public string? ISBN13 { get; set; }
        public string? Classification { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
        public short PublicationYear { get; set; }
        public string? Image { get; set; }
        public bool IsActive { get; set; }
        public short NumberOfCopies { get; set; }
        public short BorrowedCopies { get; set; }
        public bool IsAvailable { get; set; }
        public PublisherDto? Publisher { get; set; }
        public CategoryDto? Category { get; set; }
        public List<AuthorDto>? Authors { get; set; } = new();
        public List<SubCategoryDto>? SubCategories { get; set; } = new();
    }
}